include/deal.II/sundials/ida_interface.h

```diff
@@ -0,0 +1,314 @@
+                           VEC &solution_dot);
+
+    /**
+     * Clear internal memory, and start with clean objects. This function is
+     * called when the simulation start and when the user return true to a call
+     * to solver_should_restart()
+     */
+    void reset_dae(const double t,
+                   VEC &y,
+                   VEC &yp,
+                   double h,
+                   bool first_step);
+
+    /**
+     * Return a newly allocated shared_ptr<VEC>.
+     */
+    std::function<std::shared_ptr<VEC>()> create_new_vector;
+
+    /**
+     * Compute residual.
+     */
+    std::function<int(const double t,
+                      const VEC &y,
+                      const VEC &y_dot,
+                      VEC &res)> residual;
+
+    /**
+     * Compute Jacobian.
+     */
+    std::function<int(const double t,
+                      const VEC &y,
+                      const VEC &y_dot,
+                      const double alpha)> setup_jacobian;
+
+    /**
+     * Solve linear system.
+     */
+    std::function<int(const VEC &rhs, VEC &dst)> solve_jacobian_system;
+
+    /**
+     * Store solutions to file.
+     */
+    std::function<void (const double t,
+                        const VEC &sol,
+                        const VEC &sol_dot,
+                        const unsigned int step_number)> output_step;
+
+    /**
+     * Evaluate wether the solver should be restarted (for example because the
+     * number of degrees of freedom has changed).
+     *
+     * This function is supposed to perform all operations that are necessary in
+     * `sol` and `sol_dot` to make sure that the resulting vectors are
+     * consistent, and of the correct final size.
+     */
+    std::function<bool (const double t,
+                        VEC &sol,
+                        VEC &sol_dot)> solver_should_restart;
+
+    /**
+     * Return a vector whose component are 1 if the corresponding
+     * dof is differential, 0 if algebraic.
+     */
+    std::function<VEC&()> differential_components;
+
+    /**
+     * Return a vector whose components are the weights used by IDA to
+     * compute the vector norm. The implementation of this function
+     * is optional.
+     */
+    std::function<VEC&()> get_local_tolerances;
+
+
+
+    /**
+     * Set initial time equal to @p t disregarding what is written
+     * in the parameter file.
+     */
+    void set_initial_time(const double &t);
+
+  private:
+
+    /**
+     * Throw an exception when a function with the given name is not implemented.
+     */
+    DeclException1(ExcFunctionNotProvided, std::string,
+                   << "Please provide an implementation for the function \"" << arg1 << "\"");
+
+    /**
+     * This function is executed at construction time to set the
+     * std::function above to trigger an assert if they are not
+     * implemented.
+     */
+    void set_functions_to_trigger_an_assert();
+
+    /** Initial time for the DAE.*/
```

Comment:
Could u please indent it the way we do elsewhere (i.e. The function above)? Same below